plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/file/FileHiveMetastore.java

```diff
@@ -524,12 +524,10 @@ public synchronized void dropTable(String databaseName, String tableName, boolea
 
         Path tableMetadataDirectory = getTableMetadataDirectory(databaseName, tableName);
 
-        // It is safe to delete the whole meta directory for external tables and views
-        if (!table.getTableType().equals(MANAGED_TABLE.name()) || deleteData) {
```

Comment:
In the context of Iceberg tables, it is "not safe" anymore to delete the directory corresponding to the table.